File: VB/E911.Web/Controls/WebCustomUserControl.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web.UI
Imports DevExpress.Xpo
Imports DevExpress.Utils
Imports E911.Module.Editors

Partial Public Class WebCustomUserControl
	Inherits UserControl
	Implements IXpoSessionAwareControl
	Private Sub UpdateDataSource(ByVal session As Session) Implements IXpoSessionAwareControl.UpdateDataSource
		Guard.ArgumentNotNull(session, "session")
		Dim dataSource As New XpoDataSource()
		dataSource.TypeName = GetType(DevExpress.Persistent.BaseImpl.Task).FullName
		dataSource.Session = session
		grid.DataSource = dataSource
		grid.DataBind()
	End Sub
End Class

File: CS/E911.Web/Controls/WebCustomUserControl.ascx.cs
using System;
using System.Web.UI;
using DevExpress.Xpo;
using DevExpress.Utils;
using E911.Module.Editors;

public partial class WebCustomUserControl : UserControl, IXpoSessionAwareControl {
    void IXpoSessionAwareControl.UpdateDataSource(Session session) {
        Guard.ArgumentNotNull(session, "session");
        XpoDataSource dataSource = new XpoDataSource();
        dataSource.TypeName = typeof(DevExpress.Persistent.BaseImpl.Task).FullName;
        dataSource.Session = session;
        grid.DataSource = dataSource;
        grid.DataBind();
    }
}